src/es-module-shims.js

```diff
@@ -64,6 +64,11 @@ async function topLevelLoad (url, source, polyfill) {
 }
 
 async function importShim (id, parentUrl = pageBaseUrl) {
+  // Give dynamically inserted import maps a chance to be processed
+  // if `importShim()` was called synchronously after the insertion.
+  await new Promise(resolve => setTimeout(resolve, 0))
```

Comment:
If you replace this with `await featureDetectionPromise` does that work? I think that might avoid an issue for early `importShim` calls as well actually...

Reply:
I've verified that this works. Please confirm, otherwise we are good to merge.

Reply:
@guybedford I was going to test if it'll work for the subsequent insert-imports, like after `featureDetectionPromise` is already resolved awaiting it won't introduce a task-long delay that might be necessary to make sure the mutation observer is triggered and the import map is processed, wdyt?

Reply:
Awaiting a promise always introduces a task delay. But yes we should verify it is enough of a delay too.

Reply:
isn't it a microtask though?

Reply:
anyways, I'll check that real-quick now

Reply:
Ah I see what you mean, ok we may well need to retain the previous. I think featureDetectionPromise should be in here for a possible edge case though too.

Reply:
hm it works in Chrome, Safari and Firefox for me but I'm indeed a bit worried there's still a race here that might lead to  some non-deterministic outcome. So yeah I'd prefer to bring back that task delay and will also keep the featureDetectionPromise

Reply:
I'd prefer not to be superstitious to be honest... if it works then it works!

Reply:
you think? okay let's do that, we can always fix it later, we have a test now

Reply:
Thanks, yeah all codebases become collections of superstitions over time. Would be nice to avoid it unnecessarily :P